src/openrct2-ui/windows/Finances.cpp

```diff
@@ -341,1048 +260,820 @@ static constexpr const uint32_t WindowFinancesPageHoldDownWidgets[] =
-    DrawTextBasic(dpi, w->windowPos + ScreenCoordsXY{ 8, 279 }, STR_FINANCES_SUMMARY_LOAN);
-    auto ft = Formatter();
-    ft.Add<uint16_t>(gBankLoanInterestRate);
-    DrawTextBasic(dpi, w->windowPos + ScreenCoordsXY{ 167, 279 }, STR_FINANCES_SUMMARY_AT_X_PER_YEAR, ft);
-
-    // Current cash
-    ft = Formatter();
-    ft.Add<money64>(gCash);
-    rct_string_id stringId = gCash >= 0 ? STR_CASH_LABEL : STR_CASH_NEGATIVE_LABEL;
-    DrawTextBasic(dpi, w->windowPos + ScreenCoordsXY{ 8, 294 }, stringId, ft);
-
-    // Objective related financial information
-    if (gScenarioObjective.Type == OBJECTIVE_MONTHLY_FOOD_INCOME)
-    {
-        auto lastMonthProfit = finance_get_last_month_shop_profit();
-        ft = Formatter();
-        ft.Add<money64>(lastMonthProfit);
-        DrawTextBasic(
-            dpi, w->windowPos + ScreenCoordsXY{ 280, 279 }, STR_LAST_MONTH_PROFIT_FROM_FOOD_DRINK_MERCHANDISE_SALES_LABEL, ft);
-    }
-    else
+private:
+#pragma region Summary Events
+
+    void OnMouseDownSummary(rct_widgetindex widgetIndex)
     {
-        // Park value and company value
-        ft = Formatter();
-        ft.Add<money64>(gParkValue);
-        DrawTextBasic(dpi, w->windowPos + ScreenCoordsXY{ 280, 279 }, STR_PARK_VALUE_LABEL, ft);
-        ft = Formatter();
-        ft.Add<money64>(gCompanyValue);
-        DrawTextBasic(dpi, w->windowPos + ScreenCoordsXY{ 280, 294 }, STR_COMPANY_VALUE_LABEL, ft);
+        switch (widgetIndex)
+        {
+            case WIDX_LOAN_INCREASE:
+            {
+                auto newLoan = gBankLoan + MONEY(1000, 00);
+                auto gameAction = ParkSetLoanAction(newLoan);
+                GameActions::Execute(&gameAction);
+                break;
+            }
+            case WIDX_LOAN_DECREASE:
+            {
+                if (gBankLoan > 0)
+                {
+                    auto newLoan = gBankLoan - MONEY(1000, 00);
+                    auto gameAction = ParkSetLoanAction(newLoan);
+                    GameActions::Execute(&gameAction);
+                }
+                break;
+            }
+        }
     }
-}
 
-static void WindowFinancesSummaryScrollpaint(rct_window* w, rct_drawpixelinfo* dpi, int32_t scrollIndex)
-{
-    auto screenCoords = ScreenCoordsXY{ 0, TABLE_CELL_HEIGHT + 2 };
-
-    rct_widget self = w->widgets[WIDX_SUMMARY_SCROLL];
-    int32_t row_width = std::max<uint16_t>(w->scrolls[0].h_right, self.width());
-
-    // Expenditure / Income row labels
-    for (int32_t i = 0; i < static_cast<int32_t>(ExpenditureType::Count); i++)
+    void OnPrepareDrawSummary()
     {
-        // Darken every even row
-        if (i % 2 == 0)
-            gfx_fill_rect(
-                dpi,
-                { screenCoords - ScreenCoordsXY{ 0, 1 }, screenCoords + ScreenCoordsXY{ row_width, (TABLE_CELL_HEIGHT - 2) } },
-                ColourMapA[w->colours[1]].lighter | 0x1000000);
-
-        screenCoords.y += TABLE_CELL_HEIGHT;
+        auto ft = Formatter::Common();
```

Comment:
Can you add a comment that this is setting loan widgets format arguments.

As this is done via a global it is very important that nothing else uses the global format arguments until the drawing is complete.